test/src/main/java/org/apache/accumulo/test/ComprehensiveTableOperationsIT.java

```diff
@@ -0,0 +1,937 @@
+    }
+  }
+
+  @Test
+  public void test_clone() {
+    // cloning user and system tables is tested in CloneTestIT. Ensure test exists
+    assertDoesNotThrow(() -> Class.forName(CloneTestIT.class.getName()));
+  }
+
+  @Test
+  public void test_rename() {
+    // rename for user tables is tested in RenameIT. Ensure test exists
+    assertDoesNotThrow(() -> Class.forName(RenameIT.class.getName()));
+    // rename not tested for system tables. Test basic functionality here
+    var sysTables = AccumuloTable.values();
+    var tableNames = getUniqueNames(sysTables.length);
+
+    for (int i = 0; i < sysTables.length; i++) {
+      var sysTable = sysTables[i];
+      var tableName = tableNames[i];
+      assertThrows(AccumuloException.class, () -> ops.rename(sysTable.tableName(), tableName));
+    }
+  }
+
+  @Test
+  public void test_flush() throws Exception {
+    // flush for user tables and the METADATA and ROOT tables is tested in CompactionIT. Ensure
+    // test exists
+    assertDoesNotThrow(() -> Class.forName(CompactionIT.class.getName()));
+    // flush for FATE and SCAN_REF not tested for. Test basic functionality here
+    userTable = getUniqueNames(1)[0];
+    ops.create(userTable);
+    createFateTableRow(userTable);
+    createScanRefTableRow();
+
+    for (var sysTable : Set.of(AccumuloTable.FATE, AccumuloTable.SCAN_REF)) {
+      int numFilesBeforeFlush = 0;
+
+      try (TabletsMetadata tabletsMetadata = getCluster().getServerContext().getAmple()
+          .readTablets().forTable(sysTable.tableId()).build()) {
+        for (var tm : tabletsMetadata) {
+          numFilesBeforeFlush += tm.getFiles().size();
+        }
+      }
+      ops.flush(sysTable.tableName(), null, null, true);
+      // Wait until we see the new file
+      final int finalNumFilesBeforeFlush = numFilesBeforeFlush;
+      Wait.waitFor(() -> {
+        try (TabletsMetadata tabletsMetadata = getCluster().getServerContext().getAmple()
+            .readTablets().forTable(sysTable.tableId()).build()) {
+          int numFilesAfterFlush = 0;
+          for (var tm : tabletsMetadata) {
+            numFilesAfterFlush += tm.getFiles().size();
+          }
+          return numFilesAfterFlush > finalNumFilesBeforeFlush;
+        }
+      });
+    }
+  }
+
+  @Test
+  public void
+      test_setProperty_modifyProperties_removeProperty_getProperties_getTableProperties_getConfiguration()
+          throws Exception {
+    // These may be tested elsewhere across several tests, but simpler to test all here
+    userTable = getUniqueNames(1)[0];
+    ops.create(userTable);
+    // system and user tables
+    Set<String> allTables = ops.list();
+    for (String tableName : allTables) {
+      // setProperty
+      String propKey = Property.TABLE_ARBITRARY_PROP_PREFIX.getKey() + "prop";
+      String propVal = "val";
+      String newPropVal = "newval";
+      ops.setProperty(tableName, propKey, propVal);
```

Comment:
Before initially setting the prop could add a check to ensure its not already there.

Reply:
Added in c88263153459d77dad6ce76bb3c7ac4449bdbde1